test/Spec.hs

```diff
@@ -52,9 +55,23 @@ unitTests dbPool = describe "Testing: postgresql-simple-named" $ do
         , "txtVal" =? ("baz" :: ByteString)
         ]
 
+    queryWithTestValue :: IO (Either PgNamedError TestValue)
+    queryWithTestValue = runQueryWith testValueParser "SELECT ?intVal, ?intVal, ?txtVal"
+        [ "intVal" =? (42 :: Int)
+        , "txtVal" =? ("baz" :: ByteString)
+        ]
+
     run :: Sql.Query -> [NamedParam] -> IO (Either PgNamedError TestValue)
     run q params = runNamedQuery $ Pool.withResource dbPool (\conn -> queryNamed conn q params)
 
+    runQueryWith
+        :: Sql.RowParser TestValue
+        -> Sql.Query
+        -> [NamedParam]
+        -> IO (Either PgNamedError TestValue)
+    runQueryWith rowParser q params = runNamedQuery $
+        Pool.withResource dbPool (\conn -> queryWithNamed rowParser conn q params)
```

Comment:
This function has code duplication with `run`. It would be nice to reduce code duplication

Reply:
@chshersh do you mean we do something like this: 

```haskell 
run :: Sql.Query -> [NamedParam] -> IO (Either PgNamedError TestValue)
    run q params = makePool (\conn -> queryNamed conn q params)

    runQueryWith
        :: Sql.RowParser TestValue
        -> Sql.Query
        -> [NamedParam]
        -> IO (Either PgNamedError TestValue)
    runQueryWith rowParser q params = makePool (\conn -> queryWithNamed rowParser conn q params)

    makePool f = runNamedQuery $ Pool.withResource dbPool f
```
If so, I'm not sure what the type signature for `makePool` should be, since `queryNamed` and `queryWithNamed` have a different constraints.

If not, let me know how else we could reduce the code duplication

Reply:
@oishikg The code duplication can be removed even further. The following should be possible to write:

```haskell
run = callQuery queryNamed
runWith rowParser = callQuery (queryWithNamed rowParser)
```

Reply:
Thanks for pointing this out, I've implemented this in the latest commit